spec/requests/applicants/create_address_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe 'POST /v1/applicants/:applicant_id/addresses', type: :request do
+  let(:application) { create(:application, :with_applicant) }
+  let(:applicant) { application.applicant }
+  let(:application_id) { application.application_ref }
+  let(:applicant_id) { applicant.id }
+  let(:params) do
+    {
+      address_line_one: '123',
+      address_line_two: 'High Street',
+      city: 'London',
+      county: 'Greater London',
+      postcode: 'SW1H9AJ'
+    }
+  end
+  let(:post_request) do
+    -> { post "/v1/applicants/#{applicant_id}/addresses", params: params.to_json, headers: json_headers }
+  end
+
+  context 'when there is no applicant with the provided reference' do
+    let(:applicant_id) { SecureRandom.uuid }
+
+    it 'returns a 404 response' do
+      post_request.call
+
+      expect(response).to have_http_status(404)
+      expect(response.content_type).to eq('application/json')
+    end
+  end
+
+  context 'when some of the input params are not valid' do
+    let(:params) do
+      {
+        address_line_one: '',
+        address_line_two: 'High Street',
+        city: 'London',
+        county: 'Greater London',
+        postcode: '1QQa5TF1'
+      }
+    end
+
+    it 'returns a 400 response with the validation errors' do
+      post_request.call
+
+      expected_json = {
+        errors: [
+          { field: 'address_line_one', code: 'blank' },
+          { field: 'postcode', code: 'invalid' }
+        ]
+      }
+
+      expect(response).to have_http_status(400)
+      expect(response.content_type).to eq('application/json')
+      expect(response.body).to match_json_expression(expected_json)
+    end
+  end
+
+  it 'returns a successful address payload' do
+    post_request.call
+
+    expect(response).to have_http_status(201)
+    expect(response.content_type).to eql('application/json')
+    json = JSON.parse response.body
+    expect(json['address_line_one']).to eq params[:address_line_one]
```

Comment:
**[not a blocker]** 💭 Consider using the `match_json_expression` instead so that you assert the content of the whole response payload, rather than partial matches (an example can be found [here](https://github.com/ministryofjustice/laa-apply-for-legalaid-api/blob/81446e67b02b7e42692eb466b51b58bba97a463b/spec/requests/applicants/patch_applicant_spec.rb#L55-L65)).